File: src/main/scala/committee/nova/skillful/event/impl/SkillEvent.scala
package committee.nova.skillful.event.impl

import committee.nova.skillful.impl.skill.instance.SkillInstance
import net.minecraft.entity.player.ServerPlayerEntity
import net.minecraftforge.eventbus.api.Event

abstract class SkillEvent(private val player: ServerPlayerEntity, private val skill: SkillInstance) extends Event {
  def getPlayer: ServerPlayerEntity = player

  def getSkillInstance: SkillInstance = skill
}

File: src/main/java/committee/nova/skillful/common/event/impl/SkillEvent.java
package committee.nova.skillful.common.event.impl;

import committee.nova.skillful.common.skill.SkillInstance;
import net.minecraft.world.entity.player.Player;
import net.minecraftforge.eventbus.api.Event;

public class SkillEvent extends Event {
    private final Player player;
    private final SkillInstance skill;

    protected SkillEvent(Player player, SkillInstance skill) {
        this.player = player;
        this.skill = skill;
    }

    public Player getPlayer() {
        return player;
    }

    public SkillInstance getSkill() {
        return skill;
    }
}
